app/repositories/SessionRepository.scala

```diff
@@ -93,15 +93,45 @@ class DefaultSessionRepository @Inject() (config: AppConfig, component: Reactive
     )
   }
 
-  def get(key: String): Future[RequestOutcome[ProcessContext]] =
-    findAndUpdate(Json.obj("_id" -> key), Json.obj("$set" -> Json.obj(ttlExpiryFieldName -> Json.obj("$date" -> Instant.now().toEpochMilli))))
-      .map { result =>
-        result
-          .result[DefaultSessionRepository.SessionProcess]
+  def get(key: String, pageUrl: String): Future[RequestOutcome[ProcessContext]] =
+    findAndUpdate(Json.obj("_id" -> key),
+                  Json.obj(
+                    "$set" -> Json.obj(ttlExpiryFieldName -> Json.obj("$date" -> Instant.now().toEpochMilli)),
+                    "$push" -> Json.obj("pageHistory" -> pageUrl)
+                  ),
+                  fetchNewObject = false
+    ).flatMap { r =>
+          r.result[DefaultSessionRepository.SessionProcess]
           .fold {
-            logger.warn(s"Attempt to retrieve cached process from session repo with _id=$key returned no result, lastError ${result.lastError}")
-            Left(NotFoundError): RequestOutcome[ProcessContext]
-          }(r => Right(ProcessContext(r.process, r.answers)))
+            logger.warn(s"Attempt to retrieve cached process from session repo with _id=$key returned no result, lastError ${r.lastError}")
+            Future.successful(Left(NotFoundError): RequestOutcome[ProcessContext])
+          }{ sp =>
```

Comment:
IntelliJ is reporting cyclomatic complexity is too high for the method get. Perhaps this section could be moved to a separate method.

Reply:
Fixed